sympy/solvers/ode.py

```diff
@@ -5924,7 +5927,23 @@ def _remove_coefficient(expr, x):
         # then they will need to be multiplied by sufficient x to make them so.
         # This function DOES NOT do that (it doesn't even look at the
         # homogeneous equation).
-        retdict['trialset'] = _get_trial_set(expr, x)
+        temp_set = set([])
+        temp_set_2 = set([])
+        f = Function('f')
```

Comment:
Shouldn't this be pulled in as the function of the ODE. Function('f') may already be used in an unrelated way in the expression.

Reply:
> Shouldn't this be pulled in as the function of the ODE. Function('f') may already be used in an unrelated way in the expression.

sorry, I didn't understand your point

Reply:
You shouldn't create a new function object here as it might not match the original:
```
dsolve(g(x).diff(x, 2) + g(x) + exp(x))
```

Reply:
i tried this example it gives correct ans as equation

> You shouldn't create a new function object here as it might not match the original:
> 
> ```
> dsolve(g(x).diff(x, 2) + g(x) + exp(x))
> ```

i tested this example locally it works , so if remove this statement then what should i pass in `_sub_func_doit(eq_homogeneous, f(x), ts)` because i couldn't find out the alternative

Reply:
Your code assumes that the function is called f when it might not be:
```python
>>> dsolve(g(x).diff(x, 2) + g(x) - x*sin(x) + exp(I*x))
...
NotImplementedError 
```

Reply:
> Your code assumes that the function is called f when it might not be:
> 
> ```python
> >>> dsolve(g(x).diff(x, 2) + g(x) - x*sin(x) + exp(I*x))
> ...
> NotImplementedError 
> ```

so could you please suggest how should i handle this?

Reply:
@oscarbenjamin as i have fixed this so can you please review now?